Sources/SuperValidator/Validators/Phone Number/PhoneNumber.swift

```diff
@@ -0,0 +1,166 @@
+        var countryCode = ""
+        var nationalPrefix = ""
+        switch options.phoneNumberFormatType {
+        case .nanp:
+            guard phoneNumber.matches(Regex.NANPPhoneFormat) else {
+                return .failure(.invalidPhoneNumber)
+            }
+            countryCode = phoneNumber.getCountryCodeNANP()
+        case .epp:
+            guard phoneNumber.matches(Regex.EPPPhoneFormat) else {
+                return .failure(.invalidPhoneNumber)
+            }
+            countryCode = phoneNumber.getCountryCodeEPP()
+        case .international:
+            guard phoneNumber.matches(Regex.internationalPhoneFormat) else {
+                return .failure(.invalidPhoneNumber)
+            }
+            if options.whitelistPhoneNumberCountry.isNotEmpty {
+                options.whitelistPhoneNumberCountry.forEach { phoneNumberCountry in
+                    if phoneNumber.contains(phoneNumberCountry.countryCode) {
+                        countryCode = phoneNumberCountry.countryCode
+                    }
+                    if phoneNumber.contains(phoneNumberCountry.nationalPrefix) {
+                        nationalPrefix = phoneNumberCountry.nationalPrefix
+                    }
+                }
+            }
+            if options.blacklistPhoneNumberCountry.isNotEmpty {
+                options.blacklistPhoneNumberCountry.forEach { phoneNumberCountry in
+                    if phoneNumber.contains(phoneNumberCountry.countryCode) {
+                        countryCode = phoneNumberCountry.countryCode
+                    }
+                }
+            }
+        }
+        // Whitelist Country
+        if options.whitelistPhoneNumberCountry.isNotEmpty {
+            var isFoundPhoneNumberCountryCode = false
+            var isFoundPhoneNumberNationalPrefix = false
+            var isPhoneNumberLengthValid = true
+            options.whitelistPhoneNumberCountry.forEach { phoneNumberCountry in
+                // Phone Number National Prefix
+                if phoneNumberCountry.nationalPrefix == nationalPrefix {
+                    isFoundPhoneNumberNationalPrefix = true
+                }
+                // Phone Number Country Code
+                if phoneNumberCountry.countryCode == countryCode {
+                    isFoundPhoneNumberCountryCode = true
+                }
+                // Phone number max length
+                if phoneNumberCountry.maxLength < phoneNumber.count {
+                    isPhoneNumberLengthValid = false
+                }
+            }
+            if options.isUsingNationalPrefix {
+                if !isFoundPhoneNumberNationalPrefix {
+                    return .failure(.invalidNationalPrefix)
+                }
+                if !isFoundPhoneNumberCountryCode {
+                    return .success(())
+                }
+                if !isPhoneNumberLengthValid {
+                    return .failure(.invalidMaxLength)
```

Comment:
if the isFoundPhoneNumberCountryCode value is false but the isPhoneNumberLengthValid value is false as well, then we get the success result?

Reply:
Kak Ambuy, I already update the code